proof_key.go

```diff
@@ -22,6 +22,12 @@ type KeyProof interface {
 	Bytes() []byte
 }
 
+const (
+	// Used for serialization of proofs.
+	keyExistsMagicNumber = 0x50
+	keyAbsentMagicNumber = 0x51
+)
```

Comment:
You really, really don't like the type bytes, register interface in go-wire, do you? Just reproducing pieces of this functionality by hand.

Won't block it, but I think we need to standardize... so we don't have 5 different encoding mechanisms. Merge this in. Then discuss with jae and rudi about the future of type switching in serialization.

Reply:
Haha yeah this is me giving up after 1 minute of trying to figure out how to do this with go-wire. Obviously not the way to go, so let's do this properly once we figure out what we want to do.